keps/sig-auth/3257-cluster-trust-bundles/README.md

```diff
@@ -563,6 +579,34 @@ For example, if I maintain `example.com/my-signer`, I can use the following stra
   canary object first, and assess the health of the canary workloads.
 * Once I am satisfied that the change is safe, I edit the live object.
 
+### Publishing the kube-apiserver Trust Bundle
+
+Today, the trust bundle that allows verifying kube-apiserver serving certificate(s)
+at its internal endpoints is distributed into every namespace using a configMap.
+This is so that it can be mounted along with the ServiceAccount token in order
+for the workloads to be able to communicate with the kube-apiserver.
+
+To avoid the need for these configMaps in the future, a ClusterTrustBundle API
+object will be created for all clusters:
+
+```yaml
+apiVersion: v1beta1
+kind: ClusterTrustBundle
+metadata:
+  name: kubernetes.io:kube-apiserver-server:live # TODO: should we use "active" instead of "live", maybe?
+  labels:
+    kubernetes.io/cluster-trust-bundle-version: live
+spec:
+  signerName: kubernetes.io/kube-apiserver-server
+  trustBundle: "<... PEM CA ...>"
+```
+
+This object is managed by the Kubernetes Controller Manager. It serves the same
+purpose as the `kube-root-ca.crt` configMap - to verify internal kube-apiserver
+endpoints. There is currently no in-tree signer designated for these purposes,
+and so the signer with name `kubernetes.io/kube-apiserver-server` is introduced
+along with this bundle.
+
```

Comment:
Describe whether kube-controller-manager will maintain this in the existing `root-ca-certificate-publisher-controller` or in a new controller loop. I think I'd lean towards putting it in the existing controller, so that someone who is replacing with their own publisher continues to only disable one loop.

Reply:
Having it in the same controller makes sense to me, and should be hopefully quite easy, too :+1: